### PR TITLE
Dynamically set trust domain on mesh config update

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/28437

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -1140,6 +1140,15 @@ func (s *Server) fetchCARoot() map[string]string {
 func (s *Server) initMeshHandlers() {
 	log.Info("initializing mesh handlers")
 	// When the mesh config or networks change, do a full push.
+	s.environment.AddMeshHandler(func() {
```
Comment (incfly) [inline pilot/pkg/bootstrap/server.go:1143]: This handler is identitical to the handler function below, only with additiona lines of `spiffe.SetTrustDomain`.

Would it work if we just add that line to the mesh handler below?

Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go:1143]: oof, thanks. note sure how this happen, that is what I meant to do

**Review by incfly — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**